### DISCUSSION TITLE
面版與圖形程序關聯

### DISCUSSION THREAD

**Discussion by WarmBed** [TOSUN-Shanghai/TSMaster#103] [Q&A] [answered]
我想問如何在面板添加按鈕，從而執行圖形程序呢?

另外有辦法在詳細描述下圖形程序下該怎生成excel嗎?

十分感謝

**Comment by freshhope** [comment]
![image](https://github.com/TOSUN-Shanghai/TSMaster/assets/16358754/a6c810df-84fd-4b39-925f-9d9afe0b00df)

configuration:
[2023_06_15_19_25_25.zip](https://github.com/TOSUN-Shanghai/TSMaster/files/11756927/2023_06_15_19_25_25.zip)

use latest beta version


![image](https://github.com/TOSUN-Shanghai/TSMaster/assets/16358754/e972c461-f1aa-4367-9214-1bfb3bc1b852)

[2023_06_15_19_29_43.zip](https://github.com/TOSUN-Shanghai/TSMaster/files/11756960/2023_06_15_19_29_43.zip)

**Reply by WarmBed** [reply]
![image](https://github.com/TOSUN-Shanghai/TSMaster/assets/72774359/039ac1b8-8f83-420e-b4c6-f460b037b101)

我已經在執行CCode 2216,但圖形流程沒有跑呢?

![image](https://github.com/TOSUN-Shanghai/TSMaster/assets/72774359/d1789036-bccd-4c94-8cc2-6625ea7a88af)

另外選用API call沒有看到2216, 請問是為甚麼呢?

**Comment by freshhope** [comment]
注意看我给你的例程
【1】小程序需要作为小程序库安装到系统中
【2】api函数只能有一个double参数，才能识别

**Reply by WarmBed** [reply]
我的參數是const double A,  我也把小程序安裝至Mini Progaram Library，在library中執行，但在流程圖中卻沒有執行
是我少了哪些步驟嗎? 謝謝

![image](https://github.com/TOSUN-Shanghai/TSMaster/assets/72774359/5db02573-0e40-4bce-89fa-805442aa6aaf)

**Comment by freshhope** [comment]
光启动小程序库是没用的，根据你的要求，你要在面板里用按钮点击来执行图形程序，所以这个函数关联到了面板按钮的点击事件中。注意看我给你的配置

**Reply by WarmBed** [reply]
有的 我嘗試在面板 按鈕用APPcall CCode956.run_gph, 點擊後由系統消息上來看並無任何動作

![image](https://github.com/TOSUN-Shanghai/TSMaster/assets/72774359/b5c82070-6bbc-45fb-803f-5164af7a2433)
![image](https://github.com/TOSUN-Shanghai/TSMaster/assets/72774359/f19d479b-ea58-499d-bc93-442ba52db6b7)

**Comment by WarmBed** [comment, marked as answer]
![image](https://github.com/TOSUN-Shanghai/TSMaster/assets/72774359/a9b41af5-62ae-46d2-a336-f0dc65af777c)

原來是流程圖名稱沒用好
